### PR TITLE
Best tip logs in a different file

### PR DESCRIPTION
**Pull request by deepthiskumar**
Writing best tip logs to a different file `mina-best-tip.log`. This is to make it easier to collect the required logs from nodes  to determine the state to fork off of
Added a new log level `best_tip_diff` that is of  `Spam` level and  doesn't have source or module

Explain how you tested your changes here.

Checklist:

- [ ] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [ ] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [ ] All tests pass (CI will check this if you didn't)
- [ ] Serialized types are in stable-versioned modules
- [ ] Does this close issues? List them:

Closes #0000
Closes #0000

### CODE REVIEW

File: src/lib/ppx_coda/log.ml [hunk 1]
```diff
@@ -42,8 +42,9 @@ module Make (Info : Ppxinfo) = struct
     in
     let log_level_expr = pexp_ident (Located.mk log_level_id) in
     (* spam logs don't contain module, location *)
```
Comment (psteckler) [inline src/lib/ppx_coda/log.ml:44]: should comment mention best_tip_diff, besides spam?

**Comment by psteckler** [comment]
Could the same be achieved by grep'ing for the event_id associated with the structured log?

That would avoid introducing additional complexity.

**Comment by deepthiskumar** [comment]
> Could the same be achieved by grep'ing for the event_id associated with the structured log?
> 
> That would avoid introducing additional complexity.

You mean as opposed to having a separate log file?

**Review by psteckler — APPROVED** [review]

**Comment by deepthiskumar** [comment]
As discussed, coda logs are not reliable for hardfork in that we store a fixed size locally. We might not get enough best tip logs from it and may require therefore community members to save older logs. With a total of 10MB files of just best tip changes we should have a good number of states to determine the state to fork off of

**Pull request merged**